src/pudl/analysis/record_linkage/model_helpers.py

```diff
@@ -1,18 +1,134 @@
 """This module provides shared tooling that can be used by all record linkage models."""
 import importlib
+from collections.abc import Callable
 
+import mlflow
 import yaml
+from dagster import Config, op
+from pydantic import BaseModel
 
+import pudl
+from pudl.workspace.setup import PudlPaths
 
-def get_model_config(model_key: str) -> dict:
+logger = pudl.logging_helpers.get_logger(__name__)
+
+
+def get_model_config(experiment_name: str) -> dict:
     """Load model configuration from yaml file."""
     config_file = (
         importlib.resources.files("pudl.package_data.settings")
         / "record_linkage_model_config.yml"
     )
     config = yaml.safe_load(config_file.open("r"))
 
-    if not (model_config := config.get(model_key)):
-        raise RuntimeError(f"No {model_key} entry in {config_file}")
+    if not (model_config := config.get(experiment_name)):
+        raise RuntimeError(f"No {experiment_name} entry in {config_file}")
+
+    return {experiment_name: model_config}
+
+
+def flatten_model_config(model_config: dict) -> dict:
+    """Take nested dictionary defining model config and flatten for logging purposes."""
+
+    def _flatten_level(config_level: dict, param_name: str):
+        flattened_dict = {}
+        for key, val in config_level.items():
+            flattened_param = f"{param_name}.{key}"
+            if isinstance(val, dict):
+                flattened_dict |= _flatten_level(val, param_name=flattened_param)
+            else:
+                flattened_dict[flattened_param[1:]] = val
+        return flattened_dict
+
+    return _flatten_level(model_config, "")
+
+
+class ExperimentTrackerConfig(Config):
+    """Dagster config to setup experiment tracking with mlflow."""
+
+    tracking_uri: str = f"sqlite:///{PudlPaths().output_dir}/experiments.sqlite"
```

Comment:
So ultimately these experiments get logged to a SQLite DB in the output directory and then can be loaded into the mlflow UI? I think it would be helpful to have more instruction or a tutorial on how to actually look at the tracked experiments

Reply:
Yeah I realized the docstrings were quite light across the board, so I did a little overhaul. I added some info on using the UI in the `ExperimentTracker` docstring, but I could see moving that. I moved everything to `pudl/analysis/ml_tools.py` to be more generic. I could see creating a specific ml focused package within `pudl/analysis/` with submodules, at which point we could move this into something like `pudl/analysis/ml_tools/experiment_tracking.py` and put it in the top level docstring. I could also do that now, but it seemed maybe excessive to create a new package when there's not too much code actually there right now.

Reply:
Ya I think the `pudl/analysis/ml_tools.py` structure is fine for now.